# Proyecto Final
Diplomado Big Data
<br>
Facultad de Ingeniería
<br>
Universidad Nacional de Colombia
<br>
18/06/2016
<br><br>
* Mery Godoy
* Diana Galindo 
* Javier Abril

El conjunto de datos escogido para este ejercicio contiene información de clientes que han solicitado préstamos relacionados con vivienda en todo el territorio nacional. Cuenta con 69.212 registros y 40 variables. El ejercicio consistió en responder a las siguientes tres preguntas:
<br>
<br>

1. ¿En qué rango de edad se pide un crédito para compra de vivienda?
2. ¿Cuál es el top-5 de las lineas de crédito más solicitadas?
3. ¿Quiénes piden más créditos de vivienda, hombres o mujeres?
4. ¿Para qué se solicitan más créditos en cada Departamento?

#### Cargue y visualización de los datos

In [3]:
#Leer CSV del proyecto como DataFrame
customers = sqlContext.read.format('com.databricks.spark.csv').options(header='true', inferSchema='true').load('/FileStore/tables/R2_HIP_Mod_Mask_2017_2018-73655.csv')
print((customers.count(), len(customers.columns)))

(69212, 40)

In [4]:
#visualizar el contenido del csv:
display(customers)

RANGOEDAD GENERO RANGOSALARIAL ESTRATO ESTADOCIVIL NIVELESTUDIO TIPOVIVIENDA REGIONALDOMICILIO DEPARTAMENTODOMICILIO CIUDADDOMICILIO LINEACREDITO DESTINOECONOMICO TIPOMONEDA FINALIDADCREDITO REALIZAABONOSANTICIPADOS TIPOSOLICITUDCREDITO AUTORIZADESCTOLIBRANZA SUBSIDIO TIENECREDITOED INDICADORVIS PERSONASACARGO RANGOMONTOAPROBADO RANGOVLRDESEMBOLSADO RANGOSALDOTOTAL RANGOMONTOABONOANT RANGOSALDOCAPITAL RANGOVALORINICIALGARANTIA RANGOVALORACTUALGARANTIA RANGOVALORSEGURO RANGOVLRCUOTA RANGOPLAZOMESES RANGOEGRESOS RANGOTIEMPOVINCULACION RANGOLTVINICIAL RANGOLTVACTUAL RANGOINGRESOSSMMLV RANGOCUOTASPENDVSPACT RANGOCUOTASPENDIENTES RANGOCUOTASPAGADAS ENMORA null null null null null null null NO ESTABLECIDO NO ESTABLECIDO null AHORRO VOLUNTARIO CREDITOS EN PESOS COMPRA VIVIENDA PESOS null NO Individual NO NO DEFINIDO null S 3 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 03- 121 a 180 MESES SIN INFORMACI�N SIN INFORMACION 04- 0.61 a 0.8 02- 0.21 a 0.4 SIN INFORMACION 04- 0.41 a 0.6 02- 61 a 120 CUOTAS 08- 85 a 96 CUOTAS 0 null null null null null null null NO ESTABLECIDO NO ESTABLECIDO null AHORRO VOLUNTARIO CREDITOS EN UVR COMPRA VIVIENDA UVR null NO Individual NO NO DEFINIDO null N 1 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 02- > 1 SMLV & <= 2 SMLV 04- 181 a 240 MESES SIN INFORMACI�N SIN INFORMACION 04- 0.61 a 0.8 03- 0.41 a 0.6 SIN INFORMACION 06- 0.81 a 1 05- 181 a 360 CUOTAS 03- 121 a 150 CUOTAS 0 null null null null null null null NO ESTABLECIDO NO ESTABLECIDO null AHORRO VOLUNTARIO CREDITOS EN UVR COMPRA VIV. LIBERACION FNA UVR null NO IndivConyugue NO NO DEFINIDO null N 0 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES SIN INFORMACI�N SIN INFORMACION SIN INFORMACION 03- 0.41 a 0.6 SIN INFORMACION 06- 0.81 a 1 05- 181 a 360 CUOTAS 02- 61 a 120 CUOTAS 0 Entre 41 y 45 a�os M 02- > 1 SMLV & <= 2 SMLV 1 UNION LIBRE BACHILLERATO PROPIA SIN HIPOTECA BOGOTA D.C. BOGOTA BOGOTA CICLICO DECRECIENTE POR PERIODO ANUAL COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Individual NO NO DEFINIDO null S 2 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES 02- > 1 SMLV & <= 2 SMLV 11- 121 a 132 MESES 04- 0.61 a 0.8 02- 0.21 a 0.4 02- > 1 SMLV & <= 2 SMLV 04- 0.41 a 0.6 02- 61 a 120 CUOTAS 09- 97 a 108 CUOTAS 0 Entre 41 y 45 a�os M 01- <= 1 SMLV null UNION LIBRE BACHILLERATO ARRENDADA SUR NARI�O PASTO CICLICO DECRECIENTE POR PERIODO ANUAL COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Individual NO NO DEFINIDO null S 3 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES SIN INFORMACI�N 14- 157 a 168 MESES 04- 0.61 a 0.8 02- 0.21 a 0.4 SIN INFORMACION 04- 0.41 a 0.6 02- 61 a 120 CUOTAS 10- 109 a 120 CUOTAS 0 Entre 26 y 30 a�os M 01- <= 1 SMLV 2 SOLTERO TECNICO FAMILIAR CENTRO HUILA NEIVA AHORRO VOLUNTARIO CREDITOS EN UVR COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Individual NO NO DEFINIDO null S 1 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES 01- <= 1 SMLV 07- 73 a 84 MESES 04- 0.61 a 0.8 03- 0.41 a 0.6 03- > 2 SMLV & <= 3 SMLV SIN INFORMACION 05- 181 a 360 CUOTAS 04- 151 a 180 CUOTAS 1 Entre 31 y 35 a�os M 01- <= 1 SMLV 2 UNION LIBRE BACHILLERATO FAMILIAR BOGOTA D.C. BOGOTA BOGOTA CICLICO DECRECIENTE POR PERIODO ANUAL COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Conjunto NO NO DEFINIDO null N 1 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES 01- <= 1 SMLV 09- 97 a 108 MESES 04- 0.61 a 0.8 03- 0.41 a 0.6 02- > 1 SMLV & <= 2 SMLV 05- 0.61 a 0.8 04- 151

In [5]:
#Mostrar el esquema del DataFrame:
customers.printSchema()

root
-- RANGOEDAD: string (nullable = true)
-- GENERO: string (nullable = true)
-- RANGOSALARIAL: string (nullable = true)
-- ESTRATO: string (nullable = true)
-- ESTADOCIVIL: string (nullable = true)
-- NIVELESTUDIO: string (nullable = true)
-- TIPOVIVIENDA: string (nullable = true)
-- REGIONALDOMICILIO: string (nullable = true)
-- DEPARTAMENTODOMICILIO: string (nullable = true)
-- CIUDADDOMICILIO: string (nullable = true)
-- LINEACREDITO: string (nullable = true)
-- DESTINOECONOMICO: string (nullable = true)
-- TIPOMONEDA: string (nullable = true)
-- FINALIDADCREDITO: string (nullable = true)
-- REALIZAABONOSANTICIPADOS: string (nullable = true)
-- TIPOSOLICITUDCREDITO: string (nullable = true)
-- AUTORIZADESCTOLIBRANZA: string (nullable = true)
-- SUBSIDIO: string (nullable = true)
-- TIENECREDITOED: string (nullable = true)
-- INDICADORVIS: string (nullable = true)
-- PERSONASACARGO: integer (nullable = true)
-- RANGOMONTOAPROBADO: string (nullable = true)
-- RANGOVLRDESEMBOLSADO: string (nullable = true)
-- RANGOSALDOTOTAL: string (nullable = true)
-- RANGOMONTOABONOANT: string (nullable = true)
-- RANGOSALDOCAPITAL: string (nullable = true)
-- RANGOVALORINICIALGARANTIA: string (nullable = true)
-- RANGOVALORACTUALGARANTIA: string (nullable = true)
-- RANGOVALORSEGURO: string (nullable = true)
-- RANGOVLRCUOTA: string (nullable = true)
-- RANGOPLAZOMESES: string (nullable = true)
-- RANGOEGRESOS: string (nullable = true)
-- RANGOTIEMPOVINCULACION: string (nullable = true)
-- RANGOLTVINICIAL: string (nullable = true)
-- RANGOLTVACTUAL: string (nullable = true)
-- RANGOINGRESOSSMMLV: string (nullable = true)
-- RANGOCUOTASPENDVSPACT: string (nullable = true)
-- RANGOCUOTASPENDIENTES: string (nullable = true)
-- RANGOCUOTASPAGADAS: string (nullable = true)
-- ENMORA: integer (nullable = true)

In [6]:
#Contar el número de filas en el conjunto de datos:
print(customers.count())

69212

#### Exploración y preprocesamiento de los datos

In [8]:
#Ver los distintos géneros de clientes en el conjunto de datos:
display(customers.select('GENERO').distinct().collect())

GENERO F null M

In [9]:
#Ver los distintos rangos de edades de los clientes en el conjunto de datos:
display(customers.select('RANGOEDAD').distinct().collect())

RANGOEDAD Entre 36 y 40 a�os Entre 31 y 35 a�os Entre 46 y 50 a�os null Entre 61 y 65 a�os Entre 66 y 70 a�os Entre 56 y 60 a�os Entre 71 y 75 a�os Mayor a 75 a�os Entre 41 y 45 a�os Entre 26 y 30 a�os Entre 51 y 55 a�os Entre 21 y 25 a�os

In [10]:
#Recategorizar los rangos de edades
from pyspark.sql.types import IntegerType
from pyspark.sql.functions import when, lit, col

def set_range(RANGOEDAD):
  if (RANGOEDAD != RANGOEDAD) | (RANGOEDAD is None):
    return None
  
  if RANGOEDAD == 'Entre 21 y 25 a�os':
    return 1
  if RANGOEDAD == 'Entre 26 y 30 a�os':
    return 1  

  if RANGOEDAD == 'Entre 31 y 35 a�os':
    return 2
  if RANGOEDAD == 'Entre 36 y 40 a�os':
    return 2  

  if RANGOEDAD == 'Entre 41 y 45 a�os':
    return 3
  if RANGOEDAD == 'Entre 46 y 50 a�os':
    return 3

  if RANGOEDAD == 'Entre 51 y 55 a�os':
    return 4
  if RANGOEDAD == 'Entre 56 y 60 a�os':
    return 4
  
  if RANGOEDAD == 'Entre 61 y 65 a�os':
    return 5
  if RANGOEDAD == 'Entre 66 y 70 a�os':
    return 5
  
  if RANGOEDAD == 'Entre 71 y 75 a�os':
    return 6
  if RANGOEDAD == 'Mayor a 75 a�os':
    return 6
    
udfRange = udf(set_range, IntegerType())
customers = customers.withColumn('RANGOEDADREC', udfRange('RANGOEDAD'))
display(customers)

RANGOEDAD GENERO RANGOSALARIAL ESTRATO ESTADOCIVIL NIVELESTUDIO TIPOVIVIENDA REGIONALDOMICILIO DEPARTAMENTODOMICILIO CIUDADDOMICILIO LINEACREDITO DESTINOECONOMICO TIPOMONEDA FINALIDADCREDITO REALIZAABONOSANTICIPADOS TIPOSOLICITUDCREDITO AUTORIZADESCTOLIBRANZA SUBSIDIO TIENECREDITOED INDICADORVIS PERSONASACARGO RANGOMONTOAPROBADO RANGOVLRDESEMBOLSADO RANGOSALDOTOTAL RANGOMONTOABONOANT RANGOSALDOCAPITAL RANGOVALORINICIALGARANTIA RANGOVALORACTUALGARANTIA RANGOVALORSEGURO RANGOVLRCUOTA RANGOPLAZOMESES RANGOEGRESOS RANGOTIEMPOVINCULACION RANGOLTVINICIAL RANGOLTVACTUAL RANGOINGRESOSSMMLV RANGOCUOTASPENDVSPACT RANGOCUOTASPENDIENTES RANGOCUOTASPAGADAS ENMORA RANGOEDADREC null null null null null null null NO ESTABLECIDO NO ESTABLECIDO null AHORRO VOLUNTARIO CREDITOS EN PESOS COMPRA VIVIENDA PESOS null NO Individual NO NO DEFINIDO null S 3 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 03- 121 a 180 MESES SIN INFORMACI�N SIN INFORMACION 04- 0.61 a 0.8 02- 0.21 a 0.4 SIN INFORMACION 04- 0.41 a 0.6 02- 61 a 120 CUOTAS 08- 85 a 96 CUOTAS 0 null null null null null null null null NO ESTABLECIDO NO ESTABLECIDO null AHORRO VOLUNTARIO CREDITOS EN UVR COMPRA VIVIENDA UVR null NO Individual NO NO DEFINIDO null N 1 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 02- > 1 SMLV & <= 2 SMLV 04- 181 a 240 MESES SIN INFORMACI�N SIN INFORMACION 04- 0.61 a 0.8 03- 0.41 a 0.6 SIN INFORMACION 06- 0.81 a 1 05- 181 a 360 CUOTAS 03- 121 a 150 CUOTAS 0 null null null null null null null null NO ESTABLECIDO NO ESTABLECIDO null AHORRO VOLUNTARIO CREDITOS EN UVR COMPRA VIV. LIBERACION FNA UVR null NO IndivConyugue NO NO DEFINIDO null N 0 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES SIN INFORMACI�N SIN INFORMACION SIN INFORMACION 03- 0.41 a 0.6 SIN INFORMACION 06- 0.81 a 1 05- 181 a 360 CUOTAS 02- 61 a 120 CUOTAS 0 null Entre 41 y 45 a�os M 02- > 1 SMLV & <= 2 SMLV 1 UNION LIBRE BACHILLERATO PROPIA SIN HIPOTECA BOGOTA D.C. BOGOTA BOGOTA CICLICO DECRECIENTE POR PERIODO ANUAL COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Individual NO NO DEFINIDO null S 2 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES 02- > 1 SMLV & <= 2 SMLV 11- 121 a 132 MESES 04- 0.61 a 0.8 02- 0.21 a 0.4 02- > 1 SMLV & <= 2 SMLV 04- 0.41 a 0.6 02- 61 a 120 CUOTAS 09- 97 a 108 CUOTAS 0 3 Entre 41 y 45 a�os M 01- <= 1 SMLV null UNION LIBRE BACHILLERATO ARRENDADA SUR NARI�O PASTO CICLICO DECRECIENTE POR PERIODO ANUAL COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Individual NO NO DEFINIDO null S 3 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES SIN INFORMACI�N 14- 157 a 168 MESES 04- 0.61 a 0.8 02- 0.21 a 0.4 SIN INFORMACION 04- 0.41 a 0.6 02- 61 a 120 CUOTAS 10- 109 a 120 CUOTAS 0 3 Entre 26 y 30 a�os M 01- <= 1 SMLV 2 SOLTERO TECNICO FAMILIAR CENTRO HUILA NEIVA AHORRO VOLUNTARIO CREDITOS EN UVR COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Individual NO NO DEFINIDO null S 1 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES 01- <= 1 SMLV 07- 73 a 84 MESES 04- 0.61 a 0.8 03- 0.41 a 0.6 03- > 2 SMLV & <= 3 SMLV SIN INFORMACION 05- 181 a 360 CUOTAS 04- 151 a 180 CUOTAS 1 1 Entre 31 y 35 a�os M 01- <= 1 SMLV 2 UNION LIBRE BACHILLERATO FAMILIAR BOGOTA D.C. BOGOTA BOGOTA CICLICO DECRECIENTE POR PERIODO ANUAL COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Conjunto NO NO DEFINIDO null N 1 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES 01- <= 1 SMLV 09- 97 a 108 MESES 04- 0.61 a 0.8 03- 0.41 a 0.6 02- > 1 SMLV 

In [11]:
#Ver los distintos rangos de edades recategorizados de los clientes en el conjunto de datos:
display(customers.select('RANGOEDADREC').groupBy('RANGOEDADREC').count().orderBy("count",ascending=False))

RANGOEDADREC count 3 26793 2 21638 4 14194 5 3518 1 2249 6 804 null 16

#### 1.¿En qué rango de edad se pide un crédito para compra de vivienda?

In [13]:
#1.¿En qué rango de edad se pide un crédito para compra de vivienda?
edadCVDF = customers.where((col("DESTINOECONOMICO") == "COMPRA DE VIV. CREDITO CONSTRUCTOR") | (col("DESTINOECONOMICO") == "COMPRA VIV. LIBERACION FNA") | (col("DESTINOECONOMICO") == "COMPRA VIV. LIBERACION FNA CORPORACION") | (col("DESTINOECONOMICO") == "COMPRA VIVIENDA") | (col("DESTINOECONOMICO") == "COMPRA VIVIENDA CON LIBERACION") | (col("DESTINOECONOMICO") == "COMPRA VIVIENDA VIS")
)
#print(edadCVDF.count())--67003
display(edadCVDF.select('RANGOEDADREC').groupBy('RANGOEDADREC').count().orderBy("count",ascending=False))

RANGOEDADREC count 3 25943 2 21142 4 13572 5 3360 1 2193 6 777 null 16

El mayor número de solicitud de crédito se presentó en personas en Entre 41 y 50 años

#### 2.¿Cuál es el top-5 de las lineas de crédito más solicitadas?

In [16]:
#2.¿Cuál es el top-5 de las lineas de crédito más solicitadas?
display(customers.select('LINEACREDITO').groupBy('LINEACREDITO').count().orderBy
("count",ascending=False)).limit(5)

LINEACREDITO count CICLICO DECRECIENTE POR PERIODO ANUAL 43057 AHORRO VOLUNTARIO CREDITOS EN PESOS 13761 CUOTAFIJAENPESOS 6726 AHORRO VOLUNTARIO CREDITOS EN UVR 5448 REFINANCIADO CH AMOR-N 3 DEC CICLIC UVR 76 CH AMORTIZACI-N 1 CUOTA CONSTANTE EN UVR 34 DECRECIENTE CICLICO ENUVR EMPLEADOS 32 CUOTA FIJA EN PESOS ACERCASA 25 CH AMORTIZACI-N 2 CAPITAL CONSTANTE UVR 20 ARRIENDO SOCIAL CESANTIAS UVR 8 CUOTA FIJA EN UVRACERCASA 7 LEASING HABITACIONAL CES PESOS 5 GRADIENTE GEOMETRICO EN PESOS 3 ABONO CONSTANTE A CAPITALFIJO 3 CH AMORTIZACI-N 3 DECRECIENTE CICLICO UVR - COBERTURA 3 LEASING HABITACIONAL CES CTA FIJA UVR 2 CUOTA CONSTANTE EN UVR AVC 1 CAPITAL CONSTANTE EN UVR AVC 1

Las líneas de crédito más solicitadas son: _Crédito cíclico decreciente por período anual_, _Ahorro voluntario crédito en pesos_, _Crédito con cuato fija en pesos_,  _Ahorro voluntario crédito en UVR_ y _Refinancición de crédito con amortización_.

#### 3.¿Quiénes piden más créditos de vivienda, hombres o mujeres?

In [19]:
#3.¿Quiénes piden más créditos de vivienda, hombres o mujeres?
display(customers.select('GENERO').groupBy('GENERO').count().orderBy("count",ascending=False))

GENERO count M 37128 F 32068 null 16

In [20]:
37128/(37128+32068+16)*100

Out[ 33 ]: 53.64387678437266

Las mujeres solicitaron el 53.64% de los créditos

#### 4. ¿Para qué se solicitan más créditos en cada Departamento?

In [23]:
## Identificando los destinos económicos de los créditos
Destino = customers.groupBy("DESTINOECONOMICO").count()
display(Destino)

DESTINOECONOMICO count COMPRA VIV. LIBERACION FNA 1353 MEJORA RENUEVA 56 COMPRA VIVIENDA VIS 407 SIN INFORMACION EN FORMULARIO 10 COMPRA DE CARTERA HIPOTECARIA 93 ARRIENDO SOCIAL 8 LIBERACION 428 LEASING HABITACIONAL 5 REESTRUCTURACION R16806 69 CESION DE GRAVAMEN HIPOTECARIO 3 COMPRA VIVIENDA 62775 CONSTRUCCION VIVIENDA 967 CONSTRUCCION VIVIENDA VIS 6 COMPRA DE VIV. CREDITO CONSTRUCTOR 9 COMPRA VIV. LIBERACION FNA CORPORACION 16 MEJORA 435 AHORRA TU ARRIENDO 2 COMPRA VIVIENDA CON LIBERACION 2443 CAMBIO LINEA DE CREDITO 127

In [24]:
Depto = customers.groupBy("DEPARTAMENTODOMICILIO").count()
display(Depto)

DEPARTAMENTODOMICILIO count SANTANDER 2110 NO ESTABLECIDO 138 CAQUETA 485 CASANARE 435 META 2990 NARI�O 1395 AMAZONAS 85 GUAVIARE 336 CAUCA 1275 VICHADA 41 ANTIOQUIA 1362 QUINDIO 748 CHOCO 422 CALDAS 2077 BOYACA 999 PROVINCIA EXTRANJERA 2 VALLE DEL CAUCA 3407 CUNDINAMARCA 7881 TOLIMA 2582 ISLAS 24 ARAUCA 173 NORTE DE SANTANDER 1493 BOGOTA 29391 HUILA 1202 SUCRE 1190 ATLANTICO 1644 BOLIVAR 576 RISARALDA 1454 PUTUMAYO 132 MAGDALENA 786 GUAINIA 149 CORDOBA 511 GUAJIRA 822 CESAR 895

La mayoría de créditos se focaliza en compra de vivienda y se solicita en la ciudad de Bogotá.

In [26]:
DestinoDept = (customers
              .groupBy('DEPARTAMENTODOMICILIO', 'DESTINOECONOMICO')
              .agg(f.count("DESTINOECONOMICO").alias("count")))
display(DestinoDept)


DEPARTAMENTODOMICILIO DESTINOECONOMICO count CUNDINAMARCA COMPRA DE CARTERA HIPOTECARIA 14 ARAUCA CONSTRUCCION VIVIENDA 13 RISARALDA COMPRA VIVIENDA 1272 BOGOTA AHORRA TU ARRIENDO 1 SUCRE REESTRUCTURACION R16806 3 CESAR COMPRA VIV. LIBERACION FNA 3 TOLIMA REESTRUCTURACION R16806 4 BOYACA COMPRA VIVIENDA 820 CORDOBA LIBERACION 1 ANTIOQUIA CAMBIO LINEA DE CREDITO 1 MAGDALENA COMPRA VIVIENDA CON LIBERACION 52 CUNDINAMARCA CAMBIO LINEA DE CREDITO 26 META ARRIENDO SOCIAL 2 META COMPRA VIV. LIBERACION FNA 125 GUAVIARE MEJORA 1 BOLIVAR CONSTRUCCION VIVIENDA 13 PUTUMAYO MEJORA 5 CUNDINAMARCA COMPRA VIV. LIBERACION FNA 103 CESAR COMPRA VIVIENDA 774 CAUCA CONSTRUCCION VIVIENDA 55 CUNDINAMARCA ARRIENDO SOCIAL 1 TOLIMA COMPRA VIVIENDA CON LIBERACION 46 ANTIOQUIA MEJORA 17 VICHADA COMPRA VIVIENDA 33 BOGOTA REESTRUCTURACION R16806 43 ATLANTICO MEJORA RENUEVA 6 CALDAS COMPRA VIVIENDA VIS 28 CASANARE MEJORA 7 CUNDINAMARCA AHORRA TU ARRIENDO 1 NARI�O COMPRA VIVIENDA 1247 NORTE DE SANTANDER MEJORA 46 SUCRE CAMBIO LINEA DE CREDITO 3 BOLIVAR COMPRA VIVIENDA 524 BOGOTA LEASING HABITACIONAL 5 CESAR COMPRA VIVIENDA CON LIBERACION 69 ATLANTICO CAMBIO LINEA DE CREDITO 8 HUILA COMPRA VIVIENDA CON LIBERACION 39 AMAZONAS CONSTRUCCION VIVIENDA 22 CHOCO COMPRA VIV. LIBERACION FNA 10 ANTIOQUIA LIBERACION 1 CHOCO COMPRA VIVIENDA 348 CAUCA MEJORA 7 NORTE DE SANTANDER CAMBIO LINEA DE CREDITO 2 CASANARE COMPRA DE VIV. CREDITO CONSTRUCTOR 2 BOYACA MEJORA 21 CESAR CONSTRUCCION VIVIENDA 10 SUCRE COMPRA VIVIENDA CON LIBERACION 11 NARI�O COMPRA VIVIENDA CON LIBERACION 29 GUAINIA COMPRA VIV. LIBERACION FNA 13 ATLANTICO CONSTRUCCION VIVIENDA 4 GUAVIARE COMPRA VIVIENDA CON LIBERACION 8 CAQUETA COMPRA VIV. LIBERACION FNA 13 ATLANTICO COMPRA DE CARTERA HIPOTECARIA 6 HUILA COMPRA DE CARTERA HIPOTECARIA 3 BOGOTA CONSTRUCCION VIVIENDA 110 NARI�O COMPRA VIV. LIBERACION FNA 46 VALLE DEL CAUCA CESION DE GRAVAMEN HIPOTECARIO 3 CORDOBA COMPRA VIVIENDA 451 CALDAS COMPRA VIVIENDA 1821 SANTANDER MEJORA 12 CHOCO REESTRUCTURACION R16806 1 NARI�O LIBERACION 4 BOGOTA COMPRA VIVIENDA 26802 PROVINCIA EXTRANJERA COMPRA VIVIENDA 2 NORTE DE SANTANDER COMPRA VIVIENDA VIS 2 CUNDINAMARCA LIBERACION 71 CHOCO CONSTRUCCION VIVIENDA 41 CORDOBA MEJORA 14 CASANARE COMPRA VIVIENDA 372 CALDAS COMPRA VIVIENDA CON LIBERACION 161 BOYACA CONSTRUCCION VIVIENDA 85 ANTIOQUIA COMPRA VIV. LIBERACION FNA 32 CAQUETA MEJORA 7 CAQUETA COMPRA VIVIENDA 434 NORTE DE SANTANDER COMPRA VIVIENDA CON LIBERACION 20 META CONSTRUCCION VIVIENDA 6 GUAJIRA MEJORA 15 CAUCA COMPRA VIVIENDA CON LIBERACION 79 QUINDIO COMPRA VIVIENDA 701 ANTIOQUIA COMPRA DE CARTERA HIPOTECARIA 3 SANTANDER CAMBIO LINEA DE CREDITO 2 BOGOTA SIN INFORMACION EN FORMULARIO 6 GUAVIARE CAMBIO LINEA DE CREDITO 2 RISARALDA CONSTRUCCION VIVIENDA 25 ANTIOQUIA ARRIENDO SOCIAL 5 QUINDIO COMPRA VIV. LIBERACION FNA 12 QUINDIO CONSTRUCCION VIVIENDA 9 CAUCA COMPRA VIVIENDA 1072 SANTANDER COMPRA VIV. LIBERACION FNA 24 CUNDINAMARCA COMPRA VIVIENDA 7324 CASANARE COMPRA VIV. LIBERACION FNA 3 BOYACA COMPRA VIVIENDA CON LIBERACION 32 SANTANDER MEJORA RENUEVA 2 ATLANTICO COMPRA VIVIENDA 1596 CALDAS COMPRA VIV. LIBERACION FNA 51 SUCRE COMPRA VIV. LIBERACION FNA 33 CESAR COMPRA VIV. LIBERACION FNA CORPORACION 8 CHOCO CONSTRUCCION VIVIENDA VIS 6 CUNDINAMARCA MEJORA RENUEVA 6 CESAR MEJORA RENUEVA 2 NARI�O REESTRUCTURACION R16806 8 MAGDALENA LIBERACION 1 RISARALDA COMPRA VIV. LIBERACION FNA 25 AMAZONAS COMPRA VIV. LIBERACION FNA 2 MAGDALENA MEJORA RENUEVA 1 ATLANTICO LIBERACION 2 GUAJIRA REESTRUCTURACION R16806 1 CAUCA COMPRA VIVIENDA VIS 10 TOLIMA LIBERACION 4 BOGOTA LIBERACION 246 VALLE DEL CAUCA COMPRA VIVIENDA CON LIBERACION 242 GUAJIRA COMPRA VIVIENDA 696 CALDAS CONSTRUCCION VIVIENDA 6 VICHADA MEJORA 6 BOGOTA CAMBIO LINEA DE CREDITO 65 MAGDALENA COMPRA VIV. LIBERACION FNA 1 MAGDALENA CONSTRUCCION VIVIENDA 30 SANTANDER LIBERACION 24 GUAINIA CONSTRUCCION VIVIENDA 6 NORTE DE SANTANDER LIBERACION 10 META COMPRA VIVIENDA CON LIBERACION 47 BOLIVAR REESTRUCTURACION R16806 2 RISARA

In [27]:
#¿Cuál es la distribución por Departamento de acuerdo al destino económico del crédito?

import pyspark.sql.functions as f
from pyspark.sql import Window

w = Window.partitionBy('DEPARTAMENTODOMICILIO')
distr = DestinoDept.withColumn('maxcnt', f.max('count').over(w))\
    .where(f.col('count') == f.col('maxcnt'))\
    .drop('maxcnt')

display(distr)

DEPARTAMENTODOMICILIO DESTINOECONOMICO count SANTANDER COMPRA VIVIENDA 1885 NO ESTABLECIDO COMPRA VIVIENDA 117 CAQUETA COMPRA VIVIENDA 434 CASANARE COMPRA VIVIENDA 372 META COMPRA VIVIENDA 2793 NARI�O COMPRA VIVIENDA 1247 AMAZONAS COMPRA VIVIENDA 59 GUAVIARE COMPRA VIVIENDA 320 CAUCA COMPRA VIVIENDA 1072 VICHADA COMPRA VIVIENDA 33 ANTIOQUIA COMPRA VIVIENDA 1215 QUINDIO COMPRA VIVIENDA 701 CHOCO COMPRA VIVIENDA 348 CALDAS COMPRA VIVIENDA 1821 BOYACA COMPRA VIVIENDA 820 PROVINCIA EXTRANJERA COMPRA VIVIENDA 2 VALLE DEL CAUCA COMPRA VIVIENDA 3027 CUNDINAMARCA COMPRA VIVIENDA 7324 TOLIMA COMPRA VIVIENDA 2421 ISLAS COMPRA VIVIENDA 19 ARAUCA COMPRA VIVIENDA 160 NORTE DE SANTANDER COMPRA VIVIENDA 1379 BOGOTA COMPRA VIVIENDA 26802 HUILA COMPRA VIVIENDA 1084 SUCRE COMPRA VIVIENDA 1078 ATLANTICO COMPRA VIVIENDA 1596 BOLIVAR COMPRA VIVIENDA 524 RISARALDA COMPRA VIVIENDA 1272 PUTUMAYO COMPRA VIVIENDA 115 MAGDALENA COMPRA VIVIENDA 684 GUAINIA COMPRA VIVIENDA 130 CORDOBA COMPRA VIVIENDA 451 GUAJIRA COMPRA VIVIENDA 696 CESAR COMPRA VIVIENDA 774

In [28]:
from pyspark.sql.functions import *

from pyspark.ml import Pipeline
from pyspark.ml.feature import StringIndexer, OneHotEncoderEstimator
indexer = StringIndexer(inputCol="LINEACREDITO", outputCol="LINEACREDITOIndex", handleInvalid="keep")
encoder = OneHotEncoderEstimator(inputCols=["LINEACREDITOIndex"], outputCols=["LINEACREDITOCategorical"])
pipeline = Pipeline(stages=[indexer, encoder])
pipeline = pipeline.fit(customers)
customers = pipeline.transform(customers)

display(customers)


RANGOEDAD GENERO RANGOSALARIAL ESTRATO ESTADOCIVIL NIVELESTUDIO TIPOVIVIENDA REGIONALDOMICILIO DEPARTAMENTODOMICILIO CIUDADDOMICILIO LINEACREDITO DESTINOECONOMICO TIPOMONEDA FINALIDADCREDITO REALIZAABONOSANTICIPADOS TIPOSOLICITUDCREDITO AUTORIZADESCTOLIBRANZA SUBSIDIO TIENECREDITOED INDICADORVIS PERSONASACARGO RANGOMONTOAPROBADO RANGOVLRDESEMBOLSADO RANGOSALDOTOTAL RANGOMONTOABONOANT RANGOSALDOCAPITAL RANGOVALORINICIALGARANTIA RANGOVALORACTUALGARANTIA RANGOVALORSEGURO RANGOVLRCUOTA RANGOPLAZOMESES RANGOEGRESOS RANGOTIEMPOVINCULACION RANGOLTVINICIAL RANGOLTVACTUAL RANGOINGRESOSSMMLV RANGOCUOTASPENDVSPACT RANGOCUOTASPENDIENTES RANGOCUOTASPAGADAS ENMORA RANGOEDADREC LINEACREDITOIndex LINEACREDITOCategorical null null null null null null null NO ESTABLECIDO NO ESTABLECIDO null AHORRO VOLUNTARIO CREDITOS EN PESOS COMPRA VIVIENDA PESOS null NO Individual NO NO DEFINIDO null S 3 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 03- 121 a 180 MESES SIN INFORMACI�N SIN INFORMACION 04- 0.61 a 0.8 02- 0.21 a 0.4 SIN INFORMACION 04- 0.41 a 0.6 02- 61 a 120 CUOTAS 08- 85 a 96 CUOTAS 0 null 1.0 List(0, 18, List(1), List(1.0)) null null null null null null null NO ESTABLECIDO NO ESTABLECIDO null AHORRO VOLUNTARIO CREDITOS EN UVR COMPRA VIVIENDA UVR null NO Individual NO NO DEFINIDO null N 1 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 02- > 1 SMLV & <= 2 SMLV 04- 181 a 240 MESES SIN INFORMACI�N SIN INFORMACION 04- 0.61 a 0.8 03- 0.41 a 0.6 SIN INFORMACION 06- 0.81 a 1 05- 181 a 360 CUOTAS 03- 121 a 150 CUOTAS 0 null 3.0 List(0, 18, List(3), List(1.0)) null null null null null null null NO ESTABLECIDO NO ESTABLECIDO null AHORRO VOLUNTARIO CREDITOS EN UVR COMPRA VIV. LIBERACION FNA UVR null NO IndivConyugue NO NO DEFINIDO null N 0 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES SIN INFORMACI�N SIN INFORMACION SIN INFORMACION 03- 0.41 a 0.6 SIN INFORMACION 06- 0.81 a 1 05- 181 a 360 CUOTAS 02- 61 a 120 CUOTAS 0 null 3.0 List(0, 18, List(3), List(1.0)) Entre 41 y 45 a�os M 02- > 1 SMLV & <= 2 SMLV 1 UNION LIBRE BACHILLERATO PROPIA SIN HIPOTECA BOGOTA D.C. BOGOTA BOGOTA CICLICO DECRECIENTE POR PERIODO ANUAL COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Individual NO NO DEFINIDO null S 2 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES 02- > 1 SMLV & <= 2 SMLV 11- 121 a 132 MESES 04- 0.61 a 0.8 02- 0.21 a 0.4 02- > 1 SMLV & <= 2 SMLV 04- 0.41 a 0.6 02- 61 a 120 CUOTAS 09- 97 a 108 CUOTAS 0 3 0.0 List(0, 18, List(0), List(1.0)) Entre 41 y 45 a�os M 01- <= 1 SMLV null UNION LIBRE BACHILLERATO ARRENDADA SUR NARI�O PASTO CICLICO DECRECIENTE POR PERIODO ANUAL COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Individual NO NO DEFINIDO null S 3 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES SIN INFORMACI�N 14- 157 a 168 MESES 04- 0.61 a 0.8 02- 0.21 a 0.4 SIN INFORMACION 04- 0.41 a 0.6 02- 61 a 120 CUOTAS 10- 109 a 120 CUOTAS 0 3 0.0 List(0, 18, List(0), List(1.0)) Entre 26 y 30 a�os M 01- <= 1 SMLV 2 SOLTERO TECNICO FAMILIAR CENTRO HUILA NEIVA AHORRO VOLUNTARIO CREDITOS EN UVR COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Individual NO NO DEFINIDO null S 1 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES 01- <= 1 SMLV 07- 73 a 84 MESES 04- 0.61 a 0.8 03- 0.41 a 0.6 03- > 2 SMLV & <= 3 SMLV SIN INFORMACION 05- 181 a 360 CUOTAS 04- 151 a 180 CUOTAS 1 1 3.0 List(0, 18, List(3), List(1.0)) Entre 31 y 35 a�os M 01- <= 1 SMLV 2 UNION LIBRE BACHILLERATO FAMILIAR BOGOTA D.C. BOGOTA BOGOTA CICLICO DECRECIENTE POR PERIODO ANUAL COMPRA VIVIENDA UVR COMPRA VIVIENDA 

In [29]:
customers.select("LINEACREDITOCategorical").show()

+-----------------------+
LINEACREDITOCategorical|
+-----------------------+
 (18,[1],[1.0])|
 (18,[3],[1.0])|
 (18,[3],[1.0])|
 (18,[0],[1.0])|
 (18,[0],[1.0])|
 (18,[3],[1.0])|
 (18,[0],[1.0])|
 (18,[1],[1.0])|
 (18,[0],[1.0])|
 (18,[0],[1.0])|
 (18,[1],[1.0])|
 (18,[3],[1.0])|
 (18,[0],[1.0])|
 (18,[2],[1.0])|
 (18,[1],[1.0])|
 (18,[0],[1.0])|
 (18,[0],[1.0])|
 (18,[0],[1.0])|
 (18,[0],[1.0])|
 (18,[3],[1.0])|
+-----------------------+
only showing top 20 rows

In [30]:
display(customers.select("GENERO","FINALIDADCREDITO")) 

GENERO FINALIDADCREDITO null null null null null null M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F null M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M null F COMPRA VIVIENDA F COMPRA VIVIENDA CON LIBERACION F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M null F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F MEJORA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIV. LIBERACION FNA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA CON LIBERACION M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F MEJORA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M null M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F null M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M null F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIV. LIBERACION FNA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA VIS F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIV. LIBERACION FNA F null M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIV. LIBERACION FNA F COMPRA VIV. LIBERACION FNA F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M null M COMPRA VIVIENDA M null F COMPRA VIVIENDA M COMPRA VIVIENDA CON LIBERACION M COMPRA VIVIENDA M LIBERACION M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA CON LIBERACION F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA CON LIBERACION M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F null F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA F CONSTRUCCION VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA F COMPRA VIVIENDA M COMPRA VIVIENDA F CAMBIO LINEA DE CREDITO F COMPRA VIVIENDA M COMPRA VIVIENDA F COMPRA VIVIENDA VIS M COMPRA VIVIENDA M C

In [31]:
GeneroDF = customers.where((col("LINEACREDITO") == "Compra de Vivienda") | (col("GENERO") == "F")).sort(asc("GENERO"))

display(GeneroDF)

RANGOEDAD GENERO RANGOSALARIAL ESTRATO ESTADOCIVIL NIVELESTUDIO TIPOVIVIENDA REGIONALDOMICILIO DEPARTAMENTODOMICILIO CIUDADDOMICILIO LINEACREDITO DESTINOECONOMICO TIPOMONEDA FINALIDADCREDITO REALIZAABONOSANTICIPADOS TIPOSOLICITUDCREDITO AUTORIZADESCTOLIBRANZA SUBSIDIO TIENECREDITOED INDICADORVIS PERSONASACARGO RANGOMONTOAPROBADO RANGOVLRDESEMBOLSADO RANGOSALDOTOTAL RANGOMONTOABONOANT RANGOSALDOCAPITAL RANGOVALORINICIALGARANTIA RANGOVALORACTUALGARANTIA RANGOVALORSEGURO RANGOVLRCUOTA RANGOPLAZOMESES RANGOEGRESOS RANGOTIEMPOVINCULACION RANGOLTVINICIAL RANGOLTVACTUAL RANGOINGRESOSSMMLV RANGOCUOTASPENDVSPACT RANGOCUOTASPENDIENTES RANGOCUOTASPAGADAS ENMORA RANGOEDADREC LINEACREDITOIndex LINEACREDITOCategorical Entre 36 y 40 a�os F 02- > 1 SMLV & <= 2 SMLV 2 SOLTERO PREGRADO PROPIA SIN HIPOTECA CENTRO SANTANDER SAN GIL CICLICO DECRECIENTE POR PERIODO ANUAL COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Individual NO NO DEFINIDO null N 1 14- > 16 SMLV 14- > 16 SMLV 11- > 10 SMLV & <= 12 SMLV 01- <= 1 SMLV 10- > 9 SMLV & <= 10 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 03- 121 a 180 MESES 01- <= 1 SMLV 13- 145 a 156 MESES 03- 0.41 a 0.6 01- 0 a 0.2 SIN INFORMACION 02- 0 a 0.2 01- 0 a 60 CUOTAS 10- 109 a 120 CUOTAS 0 2 0.0 List(0, 18, List(0), List(1.0)) Entre 46 y 50 a�os F 01- <= 1 SMLV 2 UNION LIBRE BACHILLERATO ARRENDADA BOGOTA D.C. CUNDINAMARCA SOACHA AHORRO VOLUNTARIO CREDITOS EN PESOS COMPRA VIVIENDA PESOS COMPRA VIVIENDA SI Individual NO NO DEFINIDO null S 0 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 03- 121 a 180 MESES 01- <= 1 SMLV 11- 121 a 132 MESES 04- 0.61 a 0.8 02- 0.21 a 0.4 SIN INFORMACION 04- 0.41 a 0.6 02- 61 a 120 CUOTAS 09- 97 a 108 CUOTAS 0 3 1.0 List(0, 18, List(1), List(1.0)) Entre 31 y 35 a�os F 01- <= 1 SMLV 3 SOLTERO PREGRADO PROPIA CON HIPOTECA BOGOTA D.C. BOGOTA BOGOTA CICLICO DECRECIENTE POR PERIODO ANUAL COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Individual NO NO DEFINIDO null S 2 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES 01- <= 1 SMLV 12- 133 a 144 MESES 03- 0.41 a 0.6 02- 0.21 a 0.4 02- > 1 SMLV & <= 2 SMLV 04- 0.41 a 0.6 03- 121 a 150 CUOTAS 08- 85 a 96 CUOTAS 0 2 0.0 List(0, 18, List(0), List(1.0)) Entre 36 y 40 a�os F 01- <= 1 SMLV 1 CASADO BACHILLERATO ARRENDADA CENTRO HUILA NEIVA CICLICO DECRECIENTE POR PERIODO ANUAL COMPRA VIVIENDA UVR COMPRA VIVIENDA NO Individual NO NO DEFINIDO null S 3 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 03- 121 a 180 MESES 01- <= 1 SMLV 08- 85 a 96 MESES 03- 0.41 a 0.6 02- 0.21 a 0.4 01- <= 1 SMLV 05- 0.61 a 0.8 02- 61 a 120 CUOTAS 06- 61 a 72 CUOTAS 0 2 0.0 List(0, 18, List(0), List(1.0)) Entre 41 y 45 a�os F 02- > 1 SMLV & <= 2 SMLV 2 CASADO PREGRADO ARRENDADA BOGOTA D.C. BOGOTA BOGOTA CICLICO DECRECIENTE POR PERIODO ANUAL COMPRA VIVIENDA UVR COMPRA VIVIENDA SI Individual null CAFAM null S 0 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 03- 121 a 180 MESES 01- <= 1 SMLV 13- 145 a 156 MESES 03- 0.41 a 0.6 01- 0 a 0.2 03- > 2 SMLV & <= 3 SMLV 03- 0.21 a 0.4 01- 0 a 60 CUOTAS 11- 121 a 132 CUOTAS 1 3 0.0 List(0, 18, List(0), List(1.0)) Entre 41 y 45 a�os F 01- <= 1 SMLV 3 SOLTERO BACHILLERATO ARRENDADA SUR VALLE DEL CAUCA CALI AHORRO VOLUNTARIO CREDITOS EN UVR COMPRA VIV. LIBERACION FNA UVR COMPRA VIV. LIBERACION FNA NO Individual NO NO DEFINIDO null S null 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 14- > 16 SMLV 14- > 16 SMLV 14- > 16 SMLV 01- <= 1 SMLV 01- <= 1 SMLV 04- 181 a 240 MESES 01- <= 1 SMLV 09- 97 a 108 MESES 04- 0.61 a 0.8 SIN INFORMACION 02- > 1 SMLV & <= 2 SMLV 05- 0.61 a 0.8 05- 181 a 360 CUOTAS 05- 181 a 360 CUOTAS 0 3 3.0 List(0, 18, List(3), List(1.0)) Entre 36 y 40 a�os F 01- <= 1 SMLV null 